skimage/segmentation/chanvese.py

```diff
@@ -0,0 +1,176 @@
+import numpy as np
+from scipy.ndimage import distance_transform_cdt
+
+
+def _cv_curvature(phi):
+    """ Returns the 'curvature' of a level set 'phi'.
+    """
+    m, n = phi.shape
+    P = np.hstack((np.zeros((m, 1)), phi, np.zeros((m, 1))))
+    m, n = P.shape
+    P = np.vstack((np.zeros((1, n)), P, np.zeros((1, n))))
+    fy = P[2:, 1:-1] - P[0:-2, 1:-1]
+    fx = P[1:-1, 2:] - P[1:-1, 0:-2]
+    fyy = P[2:, 1:-1] + P[0:-2, 1:-1] - 2 * phi
+    fxx = P[1:-1, 2:] + P[1:-1, 0:-2] - 2 * phi
+    fxy = .25*(P[2:, 2:]+P[:-2, :-2]-P[:-2, 2:]-P[2:, :-2])
+    G = np.sqrt(fx**2+fy**2)
+    K = (fxx*fy**2-2*fxy*fx*fy+fyy*fx**2)/(np.power(fx**2+fy**2, 1.5)+1e-10)
+    KG = K*G
+    return KG/np.max(np.abs(KG))
+
+
+def _cv_heavyside(x, eps=1.):
+    """ Returns the result of a regularised heavyside function of the
+    input value(s).
+    """
+    return 0.5*(1.+2./np.pi*np.arctan(x/eps))
+
+
+def _cv_delta(x, eps=1.):
+    """ Returns the result of a regularised dirac function of the
+    input value(s).
+    """
+    return eps/(eps**2+x**2)
+
+
+def _cv_difference_from_average_term(U0, Hphi, lambda_pos, lambda_neg):
+    """ Returns the 'energy' contribution due to the difference from
+    the average value within a region at each point.
+    """
+    H = Hphi
+    Hinv = 1.-H
+    Hsum = np.sum(H)
+    Hinvsum = np.sum(Hinv)
+    c1 = np.sum(U0*H)
+    c2 = np.sum(U0*Hinv)
+    if(Hsum != 0):
+        c1 = c1/Hsum
+    if(Hinvsum != 0):
+        c2 = c2/Hinvsum
+    return lambda_pos*(U0-c1)**2*H+lambda_neg*(U0-c2)**2*Hinv
+
+
+def _cv_edge_length_term(phi, mu):
+    """ Returns the 'energy' contribution due to the length of the edge
+    between regions at each point, multiplied by a factor 'mu'.
+    """
+    toret = _cv_curvature(phi)
+    return mu*toret
+
+
+def _cv_energy(U0, phi, mu, lambda1, lambda2, heavyside=_cv_heavyside):
+    """ Returns the total 'energy' of the current level set function
+    """
+    H = heavyside(phi)
+    return np.sum(_cv_difference_from_average_term(U0, H, lambda1, lambda2) +
+                  _cv_edge_length_term(phi, mu))
+
+
+def _cv_signed_distance_from_zero_level(phi):
+    """ Returns a level set function corresponding to a signed distance
+    from the zero level of the input level set 'phi'.
+    """
+    return distance_transform_cdt(phi > 0)-distance_transform_cdt(phi <= 0)
+
+
+def _cv_initial_shape(U0):
+    """ Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    print U0.shape
```

Comment:
Please, remove this or Python3-ize (`from __future__ import print_function`, `print()`) and extend the output if you think it is significant. Short `(512, 512)` doesn't really makes sense to me.